lib/matplotlib/offsetbox.py

```diff
@@ -263,32 +263,22 @@ def set_offset(self, xy):
         xy : (float, float) or callable
             The (x, y) coordinates of the offset in display units. These can
             either be given explicitly as a tuple (x, y), or by providing a
-            function that converts the extent into the offset. This function
-            must have the signature::
-
-                def offset(width, height, xdescent, ydescent, renderer) \
--> (float, float)
+            function dynamically computes an offset (taking the arguments
+            passed to `.OffsetBox.get_offset`).  It is recommended, but not
+            compulsory, to make such functions take no arguments.
```

Comment:
If you don't want to deprecate and a technical limitation of the signature is too cumbersome, we have to get very explicit in the description. I don't want users starting to use arbitrary arguments.

```suggestion
            passed to `.OffsetBox.get_offset`).
            It is recommended, to make such a function take no arguments.
            
            Before version 3.6, the callable must have the signature::
            
                def offset(width, height, xdescent, ydescent, renderer) \
-> (float, float)

            For backward-compatibility, we currently accept callables with
            arbitrary arguments as long as you pass the same arguments
            in calls to `.set_offset`. This should be considered an
            implementation detail. We don't guarantee supporting arbitrary
            arguments in future versions.
```